### ISSUE TITLE
Random Forest / Decision Tree tensor op

### ISSUE THREAD

**Issue by alexander-camuto** [zkonduit/ezkl#99]
Good news ! Sklearn models can be exported to ONNX ! Unfortunately they are often represented as single node within the ONNX graph. For instance in the case of a random forest: 

![image](https://user-images.githubusercontent.com/45801863/214281400-eb13d7df-8e0e-40e7-9700-ac95b23716c2.png)

We should try (if possible) to implement decision trees and random forests as a single operation within `src/tensor/ops.rs`, such that we can easily integrate such nodes into `src/circuit/polynomial.rs` (or maybe even `src/circuit/lookup.rs` if creating tree-based algos over field elements proves to be too challenging).

**Comment by ameya-deshmukh** [comment]
@alexander-camuto is this something you're still looking at?

**Comment by alexander-camuto** [comment]
Hey @ameya-deshmukh we'd love for help on this !

**Comment by ameya-deshmukh** [comment]
@alexander-camuto awesome! Is there a TG/Discord group by any chance where we can co-ordinate on this? Would love to give it a shot

**Comment by alexander-camuto** [comment]
yeah for sure -- check out the readme of this repo :) there's a link to the telegram group
